rust/datafusion/src/execution/physical_plan/common.rs

```diff
@@ -47,12 +47,12 @@ impl RecordBatchIterator {
     }
 }
 
-impl BatchIterator for RecordBatchIterator {
+impl SendableRecordBatchReader for RecordBatchIterator {
     fn schema(&self) -> Arc<Schema> {
```

Comment:
I suggest we replace this with `SchemaRef` to hide underlying impl details.

Reply:
I'm +0 on this.  I personally like seeing the explicit type signature (once it's not very verbose) but I have no problem with this change.